src/mo_types/type.ml

```diff
@@ -1665,10 +1665,12 @@ and vars_of_binds vs bs =
 and name_of_var vs v =
   match vs with
   | [] -> v
-  | v'::vs' -> name_of_var vs' (if (fst v) = (fst v') then (fst v, snd v + 1) else v)
+  | v'::vs' -> name_of_var vs' (if fst v = fst v' then (fst v, snd v + 1) else v)
 
-and pp_bind vs ppf (v, {bound; _}) =
-  if bound = Any then
+and pp_bind vs ppf (v, {bound; sort; _}) =
+  if sort = Scope && string_of_var v = "$" then
```

Comment:
Do you need to test string_of_var?
It would be nice if we could confine the sugaring to the callsite of `can_sugar` somehow. Did you try that?

Reply:
No, I didn't. Will try now!

Reply:
I am checking for "$" because that is how all starts in the `parser.yml`. In other places it seems to be `$<func-name>` or such.

Reply:
PTAL: 93b0f849221c3f9102e2acf7cf2def8b7c3b2d0c